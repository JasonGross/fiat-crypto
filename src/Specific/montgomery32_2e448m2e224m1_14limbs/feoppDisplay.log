opp = λ var : Z.Syntax.base_type → Type,
λ '(x24, x25, x23, x21, x19, x17, x15, x13, x11, x9, x7, x5, x3, x1)%core,
uint32_t x27, uint8_t x28 = subborrow_u32(0x0, 0x0, x1);
uint32_t x30, uint8_t x31 = subborrow_u32(x28, 0x0, x3);
uint32_t x33, uint8_t x34 = subborrow_u32(x31, 0x0, x5);
uint32_t x36, uint8_t x37 = subborrow_u32(x34, 0x0, x7);
uint32_t x39, uint8_t x40 = subborrow_u32(x37, 0x0, x9);
uint32_t x42, uint8_t x43 = subborrow_u32(x40, 0x0, x11);
uint32_t x45, uint8_t x46 = subborrow_u32(x43, 0x0, x13);
uint32_t x48, uint8_t x49 = subborrow_u32(x46, 0x0, x15);
uint32_t x51, uint8_t x52 = subborrow_u32(x49, 0x0, x17);
uint32_t x54, uint8_t x55 = subborrow_u32(x52, 0x0, x19);
uint32_t x57, uint8_t x58 = subborrow_u32(x55, 0x0, x21);
uint32_t x60, uint8_t x61 = subborrow_u32(x58, 0x0, x23);
uint32_t x63, uint8_t x64 = subborrow_u32(x61, 0x0, x25);
uint32_t x66, uint8_t x67 = subborrow_u32(x64, 0x0, x24);
uint32_t x68 = cmovznz32(x67, 0x0, 0xffffffff);
uint32_t x70, uint8_t x71 = addcarryx_u32(0x0, x27, (x68 & 0xffffffff));
uint32_t x73, uint8_t x74 = addcarryx_u32(x71, x30, (x68 & 0xffffffff));
uint32_t x76, uint8_t x77 = addcarryx_u32(x74, x33, (x68 & 0xffffffff));
uint32_t x79, uint8_t x80 = addcarryx_u32(x77, x36, (x68 & 0xffffffff));
uint32_t x82, uint8_t x83 = addcarryx_u32(x80, x39, (x68 & 0xffffffff));
uint32_t x85, uint8_t x86 = addcarryx_u32(x83, x42, (x68 & 0xffffffff));
uint32_t x88, uint8_t x89 = addcarryx_u32(x86, x45, (x68 & 0xffffffff));
uint32_t x91, uint8_t x92 = addcarryx_u32(x89, x48, (x68 & 0xfffffffe));
uint32_t x94, uint8_t x95 = addcarryx_u32(x92, x51, (x68 & 0xffffffff));
uint32_t x97, uint8_t x98 = addcarryx_u32(x95, x54, (x68 & 0xffffffff));
uint32_t x100, uint8_t x101 = addcarryx_u32(x98, x57, (x68 & 0xffffffff));
uint32_t x103, uint8_t x104 = addcarryx_u32(x101, x60, (x68 & 0xffffffff));
uint32_t x106, uint8_t x107 = addcarryx_u32(x104, x63, (x68 & 0xffffffff));
uint32_t x109, uint8_t _ = addcarryx_u32(x107, x66, (x68 & 0xffffffff));
(Return x109, Return x106, Return x103, Return x100, Return x97, Return x94, Return x91, Return x88, Return x85, Return x82, Return x79, Return x76, Return x73, Return x70)
     : ∀ var : Z.Syntax.base_type → Type, expr Z.Syntax.base_type Z.Syntax.op (uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t -> uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t)

Argument scope is [function_scope]
