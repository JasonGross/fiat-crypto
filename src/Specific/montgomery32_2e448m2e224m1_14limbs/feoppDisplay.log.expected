λ x : word32 * word32 * word32 * word32 * word32 * word32 * word32 * word32 * word32 * word32 * word32 * word32 * word32 * word32,
Interp-η
(λ var : Syntax.base_type → Type,
 λ '(x25, x26, x24, x22, x20, x18, x16, x14, x12, x10, x8, x6, x4, x2)%core,
 uint32_t x28, uint8_t/*bool*/ x29 = subborrow_u32(0x0, 0x0, x2);
 uint32_t x31, uint8_t/*bool*/ x32 = subborrow_u32(x29, 0x0, x4);
 uint32_t x34, uint8_t/*bool*/ x35 = subborrow_u32(x32, 0x0, x6);
 uint32_t x37, uint8_t/*bool*/ x38 = subborrow_u32(x35, 0x0, x8);
 uint32_t x40, uint8_t/*bool*/ x41 = subborrow_u32(x38, 0x0, x10);
 uint32_t x43, uint8_t/*bool*/ x44 = subborrow_u32(x41, 0x0, x12);
 uint32_t x46, uint8_t/*bool*/ x47 = subborrow_u32(x44, 0x0, x14);
 uint32_t x49, uint8_t/*bool*/ x50 = subborrow_u32(x47, 0x0, x16);
 uint32_t x52, uint8_t/*bool*/ x53 = subborrow_u32(x50, 0x0, x18);
 uint32_t x55, uint8_t/*bool*/ x56 = subborrow_u32(x53, 0x0, x20);
 uint32_t x58, uint8_t/*bool*/ x59 = subborrow_u32(x56, 0x0, x22);
 uint32_t x61, uint8_t/*bool*/ x62 = subborrow_u32(x59, 0x0, x24);
 uint32_t x64, uint8_t/*bool*/ x65 = subborrow_u32(x62, 0x0, x26);
 uint32_t x67, uint8_t/*bool*/ x68 = subborrow_u32(x65, 0x0, x25);
 uint32_t x69 = cmovznz32(x68, 0x0, 0xffffffff);
 uint32_t x70 = (x69 & 0xffffffff);
 uint32_t x72, uint8_t/*bool*/ x73 = addcarryx_u32(0x0, x28, x70);
 uint32_t x74 = (x69 & 0xffffffff);
 uint32_t x76, uint8_t/*bool*/ x77 = addcarryx_u32(x73, x31, x74);
 uint32_t x78 = (x69 & 0xffffffff);
 uint32_t x80, uint8_t/*bool*/ x81 = addcarryx_u32(x77, x34, x78);
 uint32_t x82 = (x69 & 0xffffffff);
 uint32_t x84, uint8_t/*bool*/ x85 = addcarryx_u32(x81, x37, x82);
 uint32_t x86 = (x69 & 0xffffffff);
 uint32_t x88, uint8_t/*bool*/ x89 = addcarryx_u32(x85, x40, x86);
 uint32_t x90 = (x69 & 0xffffffff);
 uint32_t x92, uint8_t/*bool*/ x93 = addcarryx_u32(x89, x43, x90);
 uint32_t x94 = (x69 & 0xffffffff);
 uint32_t x96, uint8_t/*bool*/ x97 = addcarryx_u32(x93, x46, x94);
 uint32_t x98 = (x69 & 0xfffffffe);
 uint32_t x100, uint8_t/*bool*/ x101 = addcarryx_u32(x97, x49, x98);
 uint32_t x102 = (x69 & 0xffffffff);
 uint32_t x104, uint8_t/*bool*/ x105 = addcarryx_u32(x101, x52, x102);
 uint32_t x106 = (x69 & 0xffffffff);
 uint32_t x108, uint8_t/*bool*/ x109 = addcarryx_u32(x105, x55, x106);
 uint32_t x110 = (x69 & 0xffffffff);
 uint32_t x112, uint8_t/*bool*/ x113 = addcarryx_u32(x109, x58, x110);
 uint32_t x114 = (x69 & 0xffffffff);
 uint32_t x116, uint8_t/*bool*/ x117 = addcarryx_u32(x113, x61, x114);
 uint32_t x118 = (x69 & 0xffffffff);
 uint32_t x120, uint8_t/*bool*/ x121 = addcarryx_u32(x117, x64, x118);
 uint32_t x122 = (x69 & 0xffffffff);
 uint32_t x124, uint8_t/*bool*/ _ = addcarryx_u32(x121, x67, x122);
 (Return x124, Return x120, Return x116, Return x112, Return x108, Return x104, Return x100, Return x96, Return x92, Return x88, Return x84, Return x80, Return x76, Return x72))
x
     : word32 * word32 * word32 * word32 * word32 * word32 * word32 * word32 * word32 * word32 * word32 * word32 * word32 * word32 → ReturnType (uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t * uint32_t)
